src/agent/agent.c

```diff
@@ -1805,47 +1805,51 @@ static bool ensure_orch_address(Agent *agent) {
                 hirte_log_errorf("No manager host specified for agent '%s'", agent->name);
                 return false;
         }
 
+        char *ip_address = agent->host;
+        bool host_is_ipv4 = isIPv4Addr(ip_address);
+        bool host_is_ipv6 = isIPv4Addr(ip_address);
 
-        if (!isIPv4Addr(agent->host) && !isIPv6Addr(agent->host)) {
-                char *ip_address = malloc0_array(0, sizeof(char), INET6_ADDRSTRLEN);
-                int r = get_address(agent->host, &ip_address);
-                if (r != 0) {
-                        hirte_log_errorf("Failed to get IP address from: '%s'", agent->host);
+        _cleanup_free_ char *resolved_ip_address = NULL;
+        if (!host_is_ipv4 && !host_is_ipv6) {
+                int r = get_address(ip_address, &resolved_ip_address);
+                if (r < 0) {
+                        hirte_log_errorf(
+                                        "Failed to get IP address from host '%s': %s",
+                                        agent->host,
+                                        strerror(-r));
                         return false;
                 }
-                hirte_log_infof("Translated '%s' to '%s'", agent->host, ip_address);
-                copy_str(&agent->host, ip_address);
+                hirte_log_infof("Translated '%s' to '%s'", ip_address, resolved_ip_address);
+                ip_address = resolved_ip_address;
         }
 
-        // IPv4
-        if (isIPv4Addr(agent->host)) {
+        if (host_is_ipv4 || isIPv4Addr(ip_address)) {
                 struct sockaddr_in host;
                 memset(&host, 0, sizeof(host));
                 host.sin_family = AF_INET;
                 host.sin_port = htons(agent->port);
-                r = inet_pton(AF_INET, agent->host, &host.sin_addr);
+                r = inet_pton(AF_INET, ip_address, &host.sin_addr);
                 if (r < 1) {
-                        hirte_log_errorf("INET4: Invalid host option '%s'", agent->host);
+                        hirte_log_errorf("INET4: Invalid host option '%s'", ip_address);
                         return false;
                 }
                 agent->orch_addr = assemble_tcp_address(&host);
-        } else if (isIPv6Addr(agent->host)) { // IPv6
+        } else if (host_is_ipv6 || isIPv6Addr(ip_address)) {
                 struct sockaddr_in6 host6;
                 memset(&host6, 0, sizeof(host6));
                 host6.sin6_family = AF_INET6;
                 host6.sin6_port = htons(agent->port);
-                r = inet_pton(AF_INET6, agent->host, &host6.sin6_addr);
+                r = inet_pton(AF_INET6, ip_address, &host6.sin6_addr);
                 if (r < 1) {
-                        hirte_log_errorf("INET6: Invalid host option '%s'", agent->host);
+                        hirte_log_errorf("INET6: Invalid host option '%s'", ip_address);
                         return false;
                 }
+                // TODO: add assemble_tcp_address_v6
```

Comment:
correct.

Reply:
Added the assemble_tcp_address_v6 here since it seems to be already implemented.